src/main/java/com/iota/iri/hash/PearlDiver.java

```diff
@@ -44,179 +81,135 @@ public synchronized boolean search(final int[] transactionTrits, final int minWe
-                        midCurlStateHigh[i] = 0b1111111111111111111111111111111111111111111111111111111111111111L;
-
-                    } break;
-
-                    default: {
+            Pow pwork = new Pow(midCurlStateCopyLow, midCurlStateCopyHigh, transactionTrits, minWeightMagnitude);
+            workers[threadIndex] = new Thread(pwork);
+            workers[threadIndex].start();
+        }
 
-                        midCurlStateLow[i] = 0b1111111111111111111111111111111111111111111111111111111111111111L;
-                        midCurlStateHigh[i] = 0b0000000000000000000000000000000000000000000000000000000000000000L;
-                    }
+        for (Thread worker : workers) {
+            try {
+                worker.join();
+            } catch (final InterruptedException e) {
+                synchronized (syncObj) {
+                    state = CANCELLED;
                 }
             }
-
-            midCurlStateLow[162 + 0] = 0b1101101101101101101101101101101101101101101101101101101101101101L;
-            midCurlStateHigh[162 + 0] = 0b1011011011011011011011011011011011011011011011011011011011011011L;
-            midCurlStateLow[162 + 1] = 0b1111000111111000111111000111111000111111000111111000111111000111L;
-            midCurlStateHigh[162 + 1] = 0b1000111111000111111000111111000111111000111111000111111000111111L;
-            midCurlStateLow[162 + 2] = 0b0111111111111111111000000000111111111111111111000000000111111111L;
-            midCurlStateHigh[162 + 2] = 0b1111111111000000000111111111111111111000000000111111111111111111L;
-            midCurlStateLow[162 + 3] = 0b1111111111000000000000000000000000000111111111111111111111111111L;
-            midCurlStateHigh[162 + 3] = 0b0000000000111111111111111111111111111111111111111111111111111111L;
-
         }
 
-        if (numberOfThreads <= 0) {
-            numberOfThreads = Math.max(Runtime.getRuntime().availableProcessors() - 1, 1);
-        }
-
-        Thread[] workers = new Thread[numberOfThreads];
-
-        while (numberOfThreads-- > 0) {
-
-            final int threadIndex = numberOfThreads;
-            Thread worker = (new Thread(() -> {
-
-                final long[] midCurlStateCopyLow = new long[CURL_STATE_LENGTH], midCurlStateCopyHigh = new long[CURL_STATE_LENGTH];
-                System.arraycopy(midCurlStateLow, 0, midCurlStateCopyLow, 0, CURL_STATE_LENGTH);
-                System.arraycopy(midCurlStateHigh, 0, midCurlStateCopyHigh, 0, CURL_STATE_LENGTH);
-                for (int i = threadIndex; i-- > 0; ) {
-                    increment(midCurlStateCopyLow, midCurlStateCopyHigh, 162 + CURL_HASH_LENGTH / 9,
-                            162 + (CURL_HASH_LENGTH / 9) * 2);
+        return state == COMPLETED;
+    }
 
-                }
+    private static void para(final int[] transactionTrits, final long[] midCurlStateLow, final long[] midCurlStateHigh) {
```

Comment:
What does  `para` stand for.

Reply:
This name ```para``` origins from other pow implementation

e.g. https://github.com/iotaledger/giota/blob/fb9a0f2af32757bc6c740164a77157369fe44e56/pow_c.go#L178

The reason that I add this function here is to reduce too long methods, reported by codacy